test/deployframework/local.go

```diff
@@ -0,0 +1,154 @@
+	}
+
+	lc.Logger.Infof("Storing the metering-operator container logs to the '%s' path", logFile.Name())
+
+	lc.MeteringOperatorCmd.Stdout = logFile
+	lc.MeteringOperatorCmd.Stderr = logFile
+
+	err = lc.MeteringOperatorCmd.Start()
+	if err != nil {
+		return fmt.Errorf("failed to run the metering-operator container: %v", err)
+	}
+
+	return nil
+}
+
+// RunReportingOperatorLocal is a method that runs the reporting-operator locally
+func (lc *LocalCtx) RunReportingOperatorLocal(apiListenPort, metricsListenPort, pprofListenPort int, token string) error {
+	var err error
+
+	envVarArr := []string{
+		"METERING_NAMESPACE=" + lc.Namespace,
+		"METERING_USE_SERVICE_ACCOUNT_AS_PROM_TOKEN=false",
+		"REPORTING_OPERATOR_PROMETHEUS_BEARER_TOKEN=" + token,
+		"REPORTING_OPERATOR_API_LISTEN=" + fmt.Sprintf("%s:%d", localAddr, apiListenPort),
+		"REPORTING_OPERATOR_METRICS_LISTEN=" + fmt.Sprintf("%s:%d", localAddr, metricsListenPort),
+		"REPORTING_OPERATOR_PPROF_LISTEN=" + fmt.Sprintf("%s:%d", localAddr, pprofListenPort),
+		"REPORTING_OPERATOR_PROMETHEUS_DATASOURCE_MAX_IMPORT_BACKFILL_DURATION=15m",
+		"REPORTING_OPERATOR_PROMETHEUS_METRICS_IMPORTER_INTERVAL=30s",
+		"REPORTING_OPERATOR_PROMETHEUS_METRICS_IMPORTER_CHUNK_SIZE=5m",
+		"REPORTING_OPERATOR_PROMETHEUS_METRICS_IMPORTER_INTERVAL=5m",
+		"REPORTING_OPERATOR_PROMETHEUS_METRICS_IMPORTER_STEP_SIZE=60s",
+	}
+
+	relPath := filepath.Join(lc.HackScriptPath, runReportingOperatorLocalScriptName)
+	targetScriptDir, err := filepath.Abs(relPath)
+	if err != nil {
+		return fmt.Errorf("failed to get the absolute path for the '%s' path: %v", relPath, err)
+	}
+
+	_, err = os.Stat(targetScriptDir)
+	if err != nil {
+		return fmt.Errorf("failed to stat the '%s' path: %v", targetScriptDir, err)
+	}
+
+	lc.ReportingOperatorCmd = exec.Command(targetScriptDir)
+
+	// TODO: is os.Environ() needed?
+	lc.ReportingOperatorCmd.Env = append(os.Environ(), envVarArr...)
+
+	lc.Logger.WithFields(logrus.Fields{
+		"command":     lc.ReportingOperatorCmd.Args,
+		"environment": envVarArr,
+	}).Infof("Running the reporting-operator locally")
+
+	logFile, err := ioutil.TempFile(lc.BasePath, reportingOperatorLogName)
+	if err != nil {
+		return fmt.Errorf("failed to create the local reporting-operator log file: %v", err)
+	}
+
+	lc.Logger.Infof("Storing the hack/run-reporting-operator-local.sh logs to the '%s' path", logFile.Name())
+
+	lc.ReportingOperatorCmd.Stdout = logFile
+	lc.ReportingOperatorCmd.Stderr = logFile
+
+	err = lc.ReportingOperatorCmd.Start()
+	if err != nil {
+		return fmt.Errorf("failed to run the reporting-operator locally: %v", err)
+	}
+
+	return nil
+}
+
+// CleanupLocal removes all resources that were created while running e2e locally
+func (lc *LocalCtx) CleanupLocal() error {
+	var errArr []string
+
+	if lc.MeteringOperatorCmd != nil && lc.ReportingOperatorCmd != nil {
+		err := cleanupLocalCmds(lc.Logger, *lc.MeteringOperatorCmd, *lc.ReportingOperatorCmd)
```

Comment:
@chancez I'm on the fence about consolidating these two commands into a `*exec.Cmd` slice/array, which would be an easier nil check and not change the implementation too much.

Reply:
Updated to a list of `exec.Cmd` since last review.